vnext/Microsoft.ReactNative/IReactDispatcher.cpp

```diff
@@ -29,7 +29,21 @@ void ReactDispatcher::Post(ReactDispatcherCallback const &callback) noexcept {
 }
 
 /*static*/ IReactDispatcher ReactDispatcher::UIThreadDispatcher() noexcept {
-  return make<ReactDispatcher>(Mso::DispatchQueue::MakeCurrentThreadUIQueue());
+  static thread_local weak_ref<IReactDispatcher> *tlsWeakDispatcher{nullptr};
+  IReactDispatcher dispatcher;
+  auto queue = Mso::DispatchQueue::GetCurrentUIThreadQueue();
+  if (queue) {
+    queue.InvokeElsePost([&queue, &dispatcher ]() noexcept {
```

Comment:
Isn't there an issue if this ends up posting.  Wouldn't it be non-deterministic if dispatcher gets set before the end of the function or not?

Should this instead be a check of HasThreadAccess?

Reply:
In this context I have got the queue from the thread. It must have the thread access. I can add the HasThreadAccess to be more obvious and a comment why it is done (to access the Queue Local values).

---
In reply to: [430521560](https://github.com/microsoft/react-native-windows/pull/5007#discussion_r430521560) [](ancestors = 430521560)

Reply:
If its the same thread, why do you need to post it?

Reply:
Added the HasThreadAccess check and comments.

---
In reply to: [430588145](https://github.com/microsoft/react-native-windows/pull/5007#discussion_r430588145) [](ancestors = 430588145,430521560)